gson/src/main/java/com/google/gson/JsonParser.java

```diff
@@ -69,25 +75,46 @@ public JsonElement parse(Reader json) throws JsonIOException, JsonSyntaxExceptio
       throw new JsonSyntaxException(e);
     }
   }
-
+  
   /**
    * Returns the next value from the JSON stream as a parse tree.
    *
    * @throws JsonParseException if there is an IOException or if the specified
    *     text is not valid JSON
    * @since 1.6
    */
+  public static JsonElement parseReader(JsonReader reader) throws JsonIOException, JsonSyntaxException {
+	boolean lenient = reader.isLenient();
+	reader.setLenient(true);
+	try {
+		return Streams.parse(reader);
+	} catch (StackOverflowError e) {
+		throw new JsonParseException("Failed parsing JSON source: " + reader + " to Json", e);
+	} catch (OutOfMemoryError e) {
+		throw new JsonParseException("Failed parsing JSON source: " + reader + " to Json", e);
+	} finally {
+		reader.setLenient(lenient);
+	}  
+  }
+  
+  /**
+   * @see JsonReader.parseString
+   */
+  public JsonElement parse(String json) throws JsonSyntaxException {
+    return parseString(json);
+  }
+
+  /**
+   * @see JsonReader.parseReader

```

Comment:
Doesn't this and the other `@see JsonReader.parseReader` have to include the parameters since there are two methods with this name?

Also, shouldn't it be a hash (`#`)  sign: `@see #parseReader(Reader)`? ([javadoc](https://docs.oracle.com/javase/8/docs/technotes/tools/windows/javadoc.html#CHDDIEDI))